plugins/modeltraining/pkg/modeltraining/plugin.go

```diff
@@ -0,0 +1,85 @@
+package modeltraining
+
+import (
+	"context"
+
+	"github.com/nats-io/nats.go"
+	opensearch "github.com/opensearch-project/opensearch-go"
+	"github.com/rancher/opni/apis"
+	managementv1 "github.com/rancher/opni/pkg/apis/management/v1"
+	"github.com/rancher/opni/pkg/logger"
+	managementext "github.com/rancher/opni/pkg/plugins/apis/apiextensions/management"
+	"github.com/rancher/opni/pkg/plugins/apis/system"
+	"github.com/rancher/opni/pkg/plugins/meta"
+	"github.com/rancher/opni/pkg/util"
+	"github.com/rancher/opni/pkg/util/future"
+	"github.com/rancher/opni/pkg/util/k8sutil"
+	"github.com/rancher/opni/plugins/modeltraining/pkg/apis/modeltraining"
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ModelTrainingPlugin struct {
+	modeltraining.UnsafeModelTrainingServer
+	system.UnimplementedSystemPluginClient
+	ctx            context.Context
+	Logger         *zap.SugaredLogger
+	k8sClient      future.Future[client.Client]
+	osClient       future.Future[*opensearch.Client]
+	natsConnection future.Future[*nats.Conn]
+	kv             future.Future[nats.KeyValue]
+}
+
+func (s *ModelTrainingPlugin) UseManagementAPI(api managementv1.ManagementClient) {
+	lg := s.Logger
+	nc, err := newNatsConnection()
+	if err != nil {
+		lg.Fatal(err)
+	}
+	mgr, err := nc.JetStream()
+	if err != nil {
+		lg.Fatal(err)
+	}
+	keyValue, err := mgr.CreateKeyValue(&nats.KeyValueConfig{
```

Comment:
If the kv store already exists, will this error? Might need to check if it exists first.